.github/workflows/diagram-cache.yml

```diff
@@ -0,0 +1,54 @@
+# Copyright DB Netz AG and contributors
+# SPDX-License-Identifier: Apache-2.0
+
+name: update_capella_diagram_cache
+
+on: push
+
+env:
+  CAPELLA_DOCKER_IMAGES_REVISION: main
+
+  # Can either be no or selected
+  CAPELLA_DOCKER_IMAGES_DROPINS: selected
+
+jobs:
+  update_capella_diagram_cache:
+      runs-on: ubuntu-latest
+      steps:
+        - name: Checkout code
+          uses: actions/checkout@v3
+        - name: Set up Python 3.11
+          uses: actions/setup-python@v3
+          with:
+            python-version: 3.11
+        - name: Install capellambse
+          run:
+            pip install "capellambse[cli]@git+https://github.com/DSD-DBS/py-capellambse.git@$CAPELLAMBSE_REVISION"
+        - name: Create diagram cache
+          run: |
+            python << EOF
+            import capellambse
+            import os
+            model = capellambse.MelodyModel(
+              os.environ["ENTRYPOINT"], diagram_cache="./diagram_cache"
+            )
+            model.update_diagram_cache(
+                "ghcr.io/dsd-dbs/capella-dockerimages/capella/base:{VERSION}-${{ env.CAPELLA_DOCKER_IMAGES_DROPINS }}-dropins-${{ env.CAPELLA_DOCKER_IMAGES_REVISION }}",
+                create_index=True,
+            )
+            EOF
+        - name: Archive diagram cache
+          uses: actions/upload-artifact@v3
+          with:
+            name: diagram_cache
+            path: diagram_cache
+      env:
+        # Virtual display used to run Capella in the background.
+        # Do not modify the value!
+        DISPLAY: ":99"
```

Comment:
I'd move the value up to the script itself, just add a new step with `export DISPLAY=:99`

Reply:
Do we actually need this, when capellambse runs the container and not the whole action runs in the container?